src/native/util/os_darwin.cpp

```diff
@@ -61,7 +64,7 @@ get_supplemental_groups_by_uid(uid_t uid, std::vector<gid_t>& groups)
     int ngroups = NGROUPS_MAX;
     //for some reason on mac getgrouplist accepts an array of int instead of gid_t. so need to create a vector of int and then insert it into groups
     std::vector<int> tmp_groups(ngroups);
-    if (getgrouplist(pw->pw_name, pw->pw_gid, &tmp_groups[0], &ngroups) < 0) {
+    if (getgrouplist(pwd.pw_name, pwd.pw_gid, &tmp_groups[0], &ngroups) < 0) {
```

Comment:
I think it's better to keep using the pointer pw->pw_name to access the info. the pwd struct is only a "storage" variable, and the pointer is the actual response.